<launch>
  <!-- Launch file for hesai AT128 LiDAR -->

    <arg name="rviz" default="true" />

    <rosparam command="load" file="$(find fast_lio)/config/hesai128.yaml" />

    <param name="feature_extract_enable" type="bool" value="0"/>
    <param name="point_filter_num" type="int" value="4"/>
    <param name="max_iteration" type="int" value="3" />
    <param name="filter_size_surf" type="double" value="0.5" />
    <param name="filter_size_map" type="double" value="0.5" />
    <param name="cube_side_length" type="double" value="1000" />
    <param name="runtime_pos_log_enable" type="bool" value="0" />
    <param name="colorize_output_features" type="bool" value="true" />
    <param name="hesai_preproc_type" type="int" value="2" />
    <param name="hesai_check_disorders" type="bool" value="false" />

    <!-- <node pkg="fast_lio" type="fastlio_mapping" name="laserMapping" output="screen" />  -->
    <node pkg="fast_lio" type="run_fast_lio_node" name="laserMapping" output="screen" /> 

    <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio)/rviz_cfg/loam_livox.rviz" />
    </group>

</launch>